beetsplug/spotify.py

```diff
@@ -639,7 +642,11 @@ def track_popularity(self, track_id=None):
 
     def track_audio_features(self, track_id=None):
         """Fetch track audio features by its Spotify ID."""
-        track_data = self._handle_response(
-            requests.get, self.audio_features_url + track_id
-        )
+        try:
+            track_data = self._handle_response(
+                requests.get, self.audio_features_url + track_id)
+            track_data.raise_for_status()
```

Comment:
I don't think this will work, unfortunately… `_handle_response` returns `response.json()`, so the value will be a `dict` instead of a `Response` object.